packages/pwa-kit-react-sdk/src/ssr/universal/utils.js

```diff
@@ -51,3 +51,64 @@ export const getProxyConfigs = () => {
     // Clone to avoid accidental mutation of important configuration variables.
     return configs.map((config) => ({...config}))
 }
+
+/**
+ * Returns the express app configuration file in object form. The object resolution will
+ * be as follows (from highest to lowest priority):
+ *
+ * > {target_name}.ext
+ * > local.ext
+ * > default.ext
+ * > package.json (mobify.key)
+ *
+ * Each file marked with `ext` can optionally be terminated with `yml`, `yaml` or
+ * `json` in that priority.
+ *
+ * NOTE: This is an isomorphic function, when run on the browser the config returned will
+ * be the value serialized in the html.
+ *
+ * @returns - the application configuration object.
+ */
+/* istanbul ignore next */
+export const getConfig = () => {
+    if (onClient) {
+        return (
+            window.__CONFIG__ ||
+            JSON.parse(document.getElementById('mobify-data').innerHTML).__CONFIG__
+        )
+    }
+
+    const isRemote = Object.prototype.hasOwnProperty.call(process.env, 'AWS_LAMBDA_FUNCTION_NAME')
+    let moduleName = process?.env?.DEPLOY_TARGET || ''
+
+    // Synchronously load the `cosmiconfig` so we don't get errors from webpack in the
+    // browser bundle.
+    const _require = eval('require')
+    const {cosmiconfigSync} = _require('cosmiconfig')
```

Comment:
@bendvc We need to sync about with this bit about V2, too! There are changes on the way that I think will be problematic to resolve here.

Reply:
As discussed in one of the above [comments](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/390#discussion_r813168156) there was a spike for V2. As pointed out by Oli these few lines are the crux of what would be difficult in V2 since the notion of externals is no longer there. So we would have to think oh how to include the 'cosmiconfig' module without having it compiled for the client. These did it with these 3 lines and externals, but the same solution doesn't work in v2.